### ISSUE TITLE
In vim, merlin.py crashes on some errors

### ISSUE THREAD

**Issue by anne-pacalet** [ocaml/merlin#1100]
I recently had this (new?) error several time when saving some files:
```
Erreur détectée en traitant function <SNR>45_BufWritePostHook[4]..<SNR>45_Update
Errors[15]..<SNR>45_CacheErrors[39]..16[1]..15[24]..SyntaxCheckers_ocaml_merlin_
GetLocList[1]..merlin#ErrorLocList :
ligne    8 :
Traceback (most recent call last):
  File "<string>", line 2, in <module>
  File "XXX/opam/2.0.3/root/4.08.1/share/merlin/vim/autoload/merlin.py", line 398, in vim_loclist
    errors = command("errors")
  File "XXX/opam/2.0.3/root/4.08.1/share/merlin/vim/autoload/merlin.py", line 188, in command
    return command2(args)
  File "XXX/opam/2.0.3/root/4.08.1/share/merlin/vim/autoload/merlin.py", line 183, in command2
    raise Error(value)
merlin.Error: 'Unbound value XXX The files X.cmi\nand Y.cmi\nmake inconsistent assumptions over interface Z'
Erreur détectée en traitant function <SNR>45_BufWritePostHook[4]..<SNR>45_Update
Errors[15]..<SNR>45_CacheErrors :
ligne   39 :
E170: :endfor manquant
Erreur détectée en traitant function <SNR>45_BufWritePostHook[4]..<SNR>45_Update
Errors :
ligne   15 :
E171: :endif manquant 
```
Hope this trace will help you find the problem.

**Comment by let-def** [comment]
A similar bug has been fixed recently for OCaml 4.08, though it might be another occurrence of the same problem that we missed.
Which version of merlin do you have installed? It it is not the latest, can you try again after updating (or maybe pinning to the dev version : `opam pin add --dev merlin`).

**Comment by anne-pacalet** [comment]
I have the version 3.3.2. I think it is the latest (in opam). I cannot reproduce the error right now, but I may try with the dev version the next time I see it. Thanks for your answer.

**Comment by let-def** [comment]
Ok. The latest version is 3.3.3 (`opam update`, then `opam show merlin` should show that 3.3.3 exists). A lot of 4.08 bug were fixed in 3.3.3, so I recommend updating.

I will close this issue for now, but feel free to comment or open another issue if the problem persists in 3.3.3.
